File: UniEatsBackEnd/Models/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class Review
    {
        public int ReviewId { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public int ItemId { get; set; }

        [Range(1, 5, ErrorMessage = "Rating must be between 1 and 5.")]
        public int Rating { get; set; }

        [StringLength(100)]
        public string ReviewTitle { get; set; }

        public string ReviewText { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        // Navigation properties (User, FoodItem)
        public User User { get; set; }
        public FoodItem FoodItem { get; set; }
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/review.ts
import { User } from './user';
import { FoodItem } from './food-item';

export interface Review {
  reviewId: number; // Unique identifier for the review
  userId: number; // ID of the user who created the review
  itemId: number; // ID of the reviewed food item
  rating: number; // Rating for the item, must be between 1 and 5
  reviewTitle?: string; // Optional title for the review, max length 100
  reviewText?: string; // Optional text of the review
  createdAt?: Date; // Defaults to the current date
  user?: User | null; // Optional navigation property for the associated user
  foodItem?: FoodItem | null; // Optional navigation property for the associated food item
}
